### PR TITLE
Binary Quantization Support for pgvector HNSW Algorithm

### PR DESCRIPTION
**Pull request by Sheharyar570**
This update introduces binary vector quantization support to the `pgvector` HNSW algorithm within our benchmarking tool. The implementation includes binary vector quantization queries with an option for reranking.
### New CLI and UI Options
- **`reranking`** *(boolean)*: Enables or disables reranking for quantized queries.
- **`quantized_fetch_limit`** *(integer)*: Specifies the number of rows to fetch for distance calculations between binary vectors before reranking.
- **`reranking_metric`** *(string)*: Chooses the distance metric for reranking, with options:
  - `COSINE`
  - `L2`
  - `IP`

### CODE REVIEW

**Comment by greenhal** [comment]
This is a great addition to VectorDBBench.  My only suggestion is would be to try to reduce the large section of duplicate code in pgvector with the filtered/unfiltered sql query definition. Lines 132-194 & 195-245, this code is identical except for the where clause.

File: vectordb_bench/frontend/components/run_test/caseSelector.py [hunk 1]
```diff
@@ -103,10 +103,12 @@ def caseConfigSetting(st, dbToCaseClusterConfigs, uiCaseItem: UICaseItem, active
                 elif config.inputType == InputType.Float:
                     caseConfig[config.label] = column.number_input(
                         config.displayLabel if config.displayLabel else config.label.value,
-                        step=config.inputConfig.get("step", 0.1),
-                        min_value=config.inputConfig["min"],
-                        max_value=config.inputConfig["max"],
-                        key=key,
+                        value=config.inputConfig["value"],
+                        help=config.inputHelp,
+                    )
```
Comment (alwayslove2013) [inline vectordb_bench/frontend/components/run_test/caseSelector.py:108]: Could you please share if there is any specific reason for considering the removal of "min" / "max" / "step" / "key"?

Comment (Sheharyar570) [reply vectordb_bench/frontend/components/run_test/caseSelector.py]: Thanks for pointing out, I made a mistake here while adding a new input type. I've reverted these changes. Can you please review it again.

**Review by alwayslove2013 — APPROVED** [review]

**Pull request merged**